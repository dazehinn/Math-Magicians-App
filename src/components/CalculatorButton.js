import PropTypes from 'prop-types';

const CalculatorButton = ({ character, className, handleOnClick }) => (
  <button type="button" className={className} onClick={handleOnClick}>{character}</button>
);

CalculatorButton.propTypes = {
  character: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  handleOnClick: PropTypes.func.isRequired,
};

export default CalculatorButton;
