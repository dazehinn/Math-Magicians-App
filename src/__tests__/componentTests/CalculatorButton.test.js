

import Calculator from '../../components/calculator';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalculatorButton from '../../components/CalculatorButton';

describe('Pad Component', () => {
  test('Tree', () => {
    const tree = renderer.create(<CalculatorButton symbol="1" function1={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Click button', () => {
    const handleClick = jest.fn();
    render(<CalculatorButton symbol="1" function1={handleClick} />);
    const button = screen.getByRole('button');
    fireEvent.click(button, {});
    expect(button.textContent).toBe('1');
  });
});