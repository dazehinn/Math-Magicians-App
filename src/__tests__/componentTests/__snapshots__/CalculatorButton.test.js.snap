// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad Component Tree 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
